[dotnet profiler] Update documentation and troubleshooting guide for GA (#15084)
DataDog/documentation: content/en/profiler/profiler_troubleshooting.md

## Missing profiles in the profile search page

> If you've configured the profiler and don't see profiles in the profile search page, here are a few settings to check depending on the Operating System.

Reviewer: ```suggestion
If you've configured the profiler and don't see profiles in the profile search page, here are things to check, depending on your operating system:
```

Author: ✅ done

{{< tabs >}}

{{% tab "Linux" %}}

1. Check that the Agent is installed and running.

2. Check the profiler has been loaded from the loader log:

   1. Open the `dotnet-native-loader-dotnet-<pid>` log file in the `/var/log/datadog` folder.

   2. Look for `CorProfiler::Initialize: Continuous Profiler initialized successfully.` near the end.

>    3. If not present, enable debug logs by setting the `DD_TRACE_DEBUG` environment variable for the application.

Reviewer: By "If not present" do you mean, if that `initialized successfully` message is not present? If so, I think we should make this part of step 2, and change the first words of this sentence to "If that message is not present..."

Author: ✅ done

   4. Restart the application.

   5. Open the `dotnet-native-loader-dotnet-<pid>` log file in the `/var/log/datadog` folder.

   6. Look for the `#Profiler` entry.

   7. Check the following lines to ensure that the profiler library has been loaded:
      ```
      [...] #Profiler
      [...] PROFILER;{BD1A650D-AC5D-4896-B64F-D6FA25D6B26A};win-x64;.\Datadog.Profiler.Native.dll
      [...] PROFILER;{BD1A650D-AC5D-4896-B64F-D6FA25D6B26A};win-x86;.\Datadog.Profiler.Native.dll
      [...] DynamicDispatcherImpl::LoadConfiguration: [PROFILER] Loading: .\Datadog.Profiler.Native.dll [AbsolutePath=/opt/datadog/linux-x64/./Datadog.Tracer.Native.so]
      [...] [PROFILER] Creating a new DynamicInstance object
      [...] Load: /opt/datadog/linux-x64/./Datadog.Tracer.Native.so
      [...] GetFunction: DllGetClassObject
      [...] GetFunction: DllCanUnloadNow
      ```

3. Check the result of profiles export:

>    1. If not set already, enable debug logs by setting the `DD_TRACE_DEBUG` environment variable for the application and restart it.

Reviewer: ```suggestion
   1. Ensure debug logs are enabled by setting the `DD_TRACE_DEBUG` environment variable to `true` for the application and restart it.
```

Author: ✅ done

   2. Open the `DD-DotNet-Profiler-Native-<Application Name>-<pid>` log file in the `/var/log/datadog` folder.

>    3. Look for `libddprof error: Failed to send profile.` entries: it means that it is not possible to contact the agent.

Reviewer: ```suggestion
   3. Look for `libddprof error: Failed to send profile.` entries. This message means that it can't contact the Agent.
```

Author: ✅ done

>    4. If not present, look for `The profile was sent. Success?` entries.

Reviewer: ```suggestion
   4. If the `Failed to send profile` message is not present, look for `The profile was sent. Success?` entries.
```

Author: ✅ done

>    5. The following means the profile has been sent successfully:

Reviewer: ```suggestion
   The following message means the profile has been sent successfully:
```
Part of step 4, not a new step.

Author: ✅ done

      ```
      true, Http code: 200
      ```

>    6. Check the following other Http codes for error possible reasons:

Reviewer: ```suggestion
   6. Check the following other HTTP codes for error possible reasons:
```

Author: ✅ done

      ```
      false, Http code: XXX
      ```
        1. 403:  // TODO

        2. 500:  // TODO

{{% /tab %}}

{{% tab "Windows" %}}

1. Check that the Agent is installed and running and is visible in the Windows Services panel.

> 2. Check the profiler has been loaded from the loader log:

Author: ✅ done

   1. Open the `dotnet-native-loader-<Application Name>-<pid>` log file in the `%ProgramData%\Datadog-APM\logs\DotNet` folder.

   2. Look for `CorProfiler::Initialize: Continuous Profiler initialized successfully.` near the end.

>    2. If not present, enable debug logs by setting the `DD_TRACE_DEBUG` environment variable for the application.

Author: ✅ done

   3. Restart the application.

   4. Open the `dotnet-native-loader-<Application Name>-<pid>` log file in the `%ProgramData%\Datadog-APM\logs\DotNet` folder.

   5. Look for the `#Profiler` entry.

   6. Check the following lines to ensure that the profiler library has been loaded:
      ```
      [...] #Profiler
      [...] PROFILER;{BD1A650D-AC5D-4896-B64F-D6FA25D6B26A};win-x64;.\Datadog.Profiler.Native.dll
      [...] PROFILER;{BD1A650D-AC5D-4896-B64F-D6FA25D6B26A};win-x86;.\Datadog.Profiler.Native.dll
      [...] DynamicDispatcherImpl::LoadConfiguration: [PROFILER] Loading: .\Datadog.Profiler.Native.dll [AbsolutePath=C:\Program Files\Datadog\.NET Tracer\win-x64\Datadog.Profiler.Native.dll]
      [...] [PROFILER] Creating a new DynamicInstance object
      [...] Load: C:\Program Files\Datadog\.NET Tracer\win-x64\Datadog.Profiler.Native.dll
      [...] GetFunction: DllGetClassObject
      [...] GetFunction: DllCanUnloadNow
      ```

3. Check the result of profiles export:

>    1. If not set already, enable debug logs by setting the `DD_TRACE_DEBUG` environment variable for the application and restart it.

Reviewer: ```suggestion
   1. Ensure debug logs are enabled by setting the `DD_TRACE_DEBUG` environment variable to `true` for the application and restart it.
```

Author: Same update

1. If debug logs were not enabled in step 2.2, set the `DD_TRACE_DEBUG` environment variable to `true` for the application and restart it.

   2. Open the `DD-DotNet-Profiler-Native-<Application Name>-<pid>` log file in the `%ProgramData%\Datadog-APM\logs\DotNet` folder.

>    3. Look for `libddprof error: Failed to send profile.` entries: it means that it is not possible to contact the agent.

Author: ✅ done

>    4. If not present, look for `The profile was sent. Success?` entries.

Author: ✅ done

>    5. The following means the profile has been sent successfully:

Reviewer: ```suggestion
   The following message means the profile has been sent successfully:
```

Author: ✅ done

      ```
      true, Http code: 200
      ```

>    6. Check the following other Http codes for error possible reasons:

Author: ✅ done
